ttnn/cpp/ttnn/operations/data_movement/tilize/tilize.hpp

```diff
@@ -0,0 +1,117 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "device/tilize_op.hpp"
+#include "tt_eager/tt_dnn/op_library/run_operation.hpp"
+#include "ttnn/decorators.hpp"
+
+namespace ttnn {
+namespace operations::data_movement {
+
+struct ExecuteTilize {
+    static ttnn::Tensor execute_on_worker_thread(
+        uint8_t queue_id,
+        const ttnn::Tensor &input_tensor,
+        const std::optional<MemoryConfig> &memory_config = std::nullopt,
+        std::optional<const DataType> output_dtype = std::nullopt,
+        bool use_multicore = false) {
+        return operation::run(
+                   Tilize{
+                       memory_config.value_or(input_tensor.memory_config()),
+                       output_dtype.value_or(input_tensor.get_dtype()),
+                       use_multicore},
+                   {input_tensor},
+                   {},
+                   {},
+                   queue_id)
+            .at(0);
+    }
+
+    static ttnn::Tensor execute_on_worker_thread(
+        const ttnn::Tensor &input_tensor,
+        const std::optional<MemoryConfig> &memory_config = std::nullopt,
+        std::optional<const DataType> output_dtype = std::nullopt,
+        bool use_multicore = false) {
```

Comment:
We see this pattern often, but I still don't understand why we need to provide this explicitly. 
Wdyt?

Reply:
@ayerofieiev-tt if we add a special type for `queue_id`, we can make it so that we only need to define the method with `queue_id` and just simply provide a default value in the infra

Reply:
I followed the existing pattern. If we clean it up, let's do it all at once in a dedicated PR.